cipher-asymmetric/src/main/scala/tsec/cipher/asymmetric/imports/imports.scala

```diff
@@ -0,0 +1,48 @@
+package tsec.cipher.asymmetric
+
+import java.security.{PrivateKey => JPrivateKey, PublicKey => JPublicKey}
+
+import cats.Id
+import cats.evidence.Is
+import tsec.common.CryptoTag
+
+package object imports {
+
+  protected[tsec] case class AsymmetricAlgorithm[T](algorithm: String, keyLength: Int) extends CryptoTag[T]
+
+  sealed case class KeyPair[A](privateKey: PrivateKey[A], publicKey: PublicKey[A])
+  object KeyPair {
+    def apply[A](jPrivateKey: JPrivateKey, jPublicKey: JPublicKey): KeyPair[A] =
+      new KeyPair[A](PrivateKey[A](jPrivateKey), PublicKey[A](jPublicKey))
+  }
+
+  sealed trait TaggedKey[T] {
+    type KeyRepr[A] <: Id[T]
```

Comment:
Just following up on our convo:
Remember, you cannot carry proof that KeyRepr[A] === T

Otherwise, the compiler reduces it down immediately, and the type segregation we want is lost.

Please separate taggedkey into two things: PrivateKey and PublicKey (alternatively copy it from signature).